packages/react-dom/src/events/ReactBrowserEventEmitter.js

```diff
@@ -118,27 +125,27 @@ export function listenTo(registrationName, contentDocumentHandle) {
   for (let i = 0; i < dependencies.length; i++) {
     const dependency = dependencies[i];
     if (!(isListening.hasOwnProperty(dependency) && isListening[dependency])) {
-      if (dependency === 'topScroll') {
-        trapCapturedEvent('topScroll', 'scroll', mountAt);
-      } else if (dependency === 'topFocus' || dependency === 'topBlur') {
-        trapCapturedEvent('topFocus', 'focus', mountAt);
-        trapCapturedEvent('topBlur', 'blur', mountAt);
+      if (dependency === TOP_SCROLL) {
+        trapCapturedEvent(TOP_SCROLL, mountAt);
+      } else if (dependency === TOP_FOCUS || dependency === TOP_BLUR) {
+        trapCapturedEvent(TOP_FOCUS, mountAt);
+        trapCapturedEvent(TOP_BLUR, mountAt);
 
         // to make sure blur and focus event listeners are only attached once
-        isListening.topBlur = true;
-        isListening.topFocus = true;
-      } else if (dependency === 'topCancel') {
+        isListening[TOP_BLUR] = true;
+        isListening[TOP_FOCUS] = true;
+      } else if (dependency === TOP_CANCEL) {
         if (isEventSupported('cancel', true)) {
-          trapCapturedEvent('topCancel', 'cancel', mountAt);
+          trapCapturedEvent(TOP_CANCEL, mountAt);
         }
-        isListening.topCancel = true;
-      } else if (dependency === 'topClose') {
+        isListening[TOP_CANCEL] = true;
+      } else if (dependency === TOP_CLOSE) {
         if (isEventSupported('close', true)) {
-          trapCapturedEvent('topClose', 'close', mountAt);
+          trapCapturedEvent(TOP_CLOSE, mountAt);
         }
-        isListening.topClose = true;
-      } else if (topLevelTypes.hasOwnProperty(dependency)) {
-        trapBubbledEvent(dependency, topLevelTypes[dependency], mountAt);
+        isListening[TOP_CLOSE] = true;
+      } else if (getRawEventName(dependency)) {
```

Comment:
Please be specific, e.g. `!== null`

Reply:
Actually, isn't this just an identity function? In this case it would always choose this code path.

Does this mean we can just remove this check (and count on nobody using unknown dependency types if they access private API)?

Maybe we could validate them in development, but not in production?